internal/testrunner/coberturacoverage.go

```diff
@@ -0,0 +1,259 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package testrunner
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"path/filepath"
+	"sort"
+)
+
+const coverageDtd = `<!DOCTYPE coverage SYSTEM "http://cobertura.sourceforge.net/xml/coverage-04.dtd">`
+
+// CoberturaCoverage is the root element for a Cobertura XML report.
+type CoberturaCoverage struct {
+	XMLName         xml.Name            `xml:"coverage"`
+	LineRate        float32             `xml:"line-rate,attr"`
+	BranchRate      float32             `xml:"branch-rate,attr"`
+	Version         string              `xml:"version,attr"`
+	Timestamp       int64               `xml:"timestamp,attr"`
+	LinesCovered    int64               `xml:"lines-covered,attr"`
+	LinesValid      int64               `xml:"lines-valid,attr"`
+	BranchesCovered int64               `xml:"branches-covered,attr"`
+	BranchesValid   int64               `xml:"branches-valid,attr"`
+	Complexity      float32             `xml:"complexity,attr"`
+	Sources         []*CoberturaSource  `xml:"sources>source"`
+	Packages        []*CoberturaPackage `xml:"packages>package"`
+}
+
+// CoberturaSource represents a base path to the covered source code.
+type CoberturaSource struct {
+	Path string `xml:",chardata"`
+}
+
+// CoberturaPackage represents a package in a Cobertura XML report.
+type CoberturaPackage struct {
+	Name       string            `xml:"name,attr"`
+	LineRate   float32           `xml:"line-rate,attr"`
+	BranchRate float32           `xml:"branch-rate,attr"`
+	Complexity float32           `xml:"complexity,attr"`
+	Classes    []*CoberturaClass `xml:"classes>class"`
+}
+
+// CoberturaClass represents a class in a Cobertura XML report.
+type CoberturaClass struct {
+	Name       string             `xml:"name,attr"`
+	Filename   string             `xml:"filename,attr"`
+	LineRate   float32            `xml:"line-rate,attr"`
+	BranchRate float32            `xml:"branch-rate,attr"`
+	Complexity float32            `xml:"complexity,attr"`
+	Methods    []*CoberturaMethod `xml:"methods>method"`
+	Lines      []*CoberturaLine   `xml:"lines>line"`
+}
+
+// CoberturaMethod represents a method in a Cobertura XML report.
+type CoberturaMethod struct {
+	Name       string           `xml:"name,attr"`
+	Signature  string           `xml:"signature,attr"`
+	LineRate   float32          `xml:"line-rate,attr"`
+	BranchRate float32          `xml:"branch-rate,attr"`
+	Complexity float32          `xml:"complexity,attr"`
+	Lines      []*CoberturaLine `xml:"lines>line"`
+}
```

Comment:
Removed Hits field from this struct since it is not present in the DTD

Reply:
I just checked the Jenkins plugin code , looking at the code it does not look like it was used. And their tests do not include that field neither 

https://github.com/jenkinsci/cobertura-plugin/blob/ff0973461f60c28dbc0bf5ab4853a5aec4798210/src/test/resources/hudson/plugins/cobertura/coverage-with-data.xml